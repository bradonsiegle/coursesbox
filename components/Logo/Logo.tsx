import styled from "@emotion/styled";
import { css } from "@emotion/react";

export type LogoProps = { /** Logo size in rem */ size?: number };

export const Logo = styled.header<LogoProps>`
  font-family: Monoton;
  font-size: ${({ size = 3 }) => `${size}rem`};
  ${({ theme, size = 3 }) => {
    return css`
      color: ${theme.font.logo};
      text-shadow: 0 0 0.3rem ${theme.font.logoShadow1},
        0 0 ${0.05 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.07 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.08 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.1 * size}rem ${theme.font.logoShadow2};
    `;
  }}
`;
