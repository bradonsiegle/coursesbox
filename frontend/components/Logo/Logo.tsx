import styled from "@emotion/styled";
import { css } from "@emotion/react";

export type LogoProps = { /** Logo size in rem */ size?: number };

export const Logo = styled.header<LogoProps>`
  font-family: "Playfair Display", serif;
  font-size: ${({ size = 3 }) => `${size}rem`};
  @media (min-width: 768px) {
    font-size: ${({ size = 3 }) => `${size + 0.4}rem`};
  }
  ${({ theme, size = 3 }) => {
    return css`
      color: ${theme.font.logo};
      text-shadow: 0 0 0.3rem ${theme.font.logoShadow1},
        0 0 ${0.05 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.07 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.08 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.1 * size}rem ${theme.font.logoShadow2};
    `;
  }}
`;
