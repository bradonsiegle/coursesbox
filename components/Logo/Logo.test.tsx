import { render, screen } from '@/test-utils';

import { Logo } from './Logo';

describe('Logo test cases', () => {
	it('Should render logo', () => {
		const { asFragment } = render(<Logo>CoursesBox</Logo>);
		expect(asFragment()).toMatchSnapshot();
	});
	it('Should render logo with custom size', () => {
		const { asFragment } = render(<Logo size={5}>CoursesBox</Logo>);
		expect(asFragment()).toMatchSnapshot();
	});
});
