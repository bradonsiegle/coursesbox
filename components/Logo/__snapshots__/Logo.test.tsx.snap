// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo test cases Should render logo 1`] = `
<DocumentFragment>
  <header
    class="css-1k5lxc1"
  >
    CoursesBox
  </header>
</DocumentFragment>
`;

exports[`Logo test cases Should render logo with custom size 1`] = `
<DocumentFragment>
  <header
    class="css-1xr678i"
    size="5"
  >
    CoursesBox
  </header>
</DocumentFragment>
`;
